src/Compiler/CodeGen/IlxGen.fs

```diff
@@ -11555,18 +11604,19 @@ let CodegenAssembly cenv eenv mgbuf implFiles =
     | None -> ()
     | Some (firstImplFiles, lastImplFile) ->
 
-        // Generate the assembly sequentially, implementation file by implementation file.
-        //
-        // NOTE: In theory this could be done in parallel, except for the presence of linear
-        // state in the AssemblyBuilder
         let eenv = List.fold (GenImplFile cenv mgbuf None) eenv firstImplFiles
         let eenv = GenImplFile cenv mgbuf cenv.options.mainMethodInfo eenv lastImplFile
 
+        eenv.delayedFileGenReverse
+        |> Array.ofList
+        |> Array.rev
+        |> ArrayParallel.iter (fun genMeths -> genMeths |> Array.iter (fun gen -> gen ()))
```

Comment:
This could probably be slightly simplified

```suggestion
        |> ArrayParallel.iter (Array.iter (fun gen -> gen ()))
```

...or dramatically simplified, but we probably don't want that :) 

```suggestion
        |> ArrayParallel.iter (Array.iter ((|>) ()))
```

Reply:
Even though it is less characters to read, I am opting in for explicit lambdas here to make it obvious that this is a collection of functions and each function is being invoked.

(to make it really stand out opposite to classical approach of collections holding data which is passed around)